File: apps/minecraft/velocity/src/main/java/crabcraft/net/crabUtilities/velocity/RedisPools.java
package crabcraft.net.crabUtilities.velocity;

import redis.clients.jedis.JedisPool;
import redis.clients.jedis.JedisPoolConfig;

public final class RedisPools {

    private static final int TIMEOUT_MS = 2_000;

    private RedisPools() {}

    public static JedisPool create(VelocityConfig config) {
        return create(config, new JedisPoolConfig());
    }

    public static JedisPool create(VelocityConfig config, int maxTotal) {
        JedisPoolConfig poolConfig = new JedisPoolConfig();
        poolConfig.setMaxTotal(maxTotal);
        return create(config, poolConfig);
    }

    public static JedisPool create(VelocityConfig config, JedisPoolConfig poolConfig) {
        String password = config.getRedisPassword();
        if (password != null && !password.isEmpty()) {
            return new JedisPool(poolConfig, config.getRedisHost(),
                    config.getRedisPort(), TIMEOUT_MS, password);
        }
        return new JedisPool(poolConfig, config.getRedisHost(),
                config.getRedisPort(), TIMEOUT_MS);
    }
}

File: apps/minecraft/velocity/src/main/kotlin/crabcraft/net/crabUtilities/velocity/RedisPools.kt
package crabcraft.net.crabUtilities.velocity

import redis.clients.jedis.JedisPool
import redis.clients.jedis.JedisPoolConfig

object RedisPools {
    private const val TIMEOUT_MS = 2_000

    @JvmStatic fun create(config: VelocityConfig): JedisPool = create(config, JedisPoolConfig())

    @JvmStatic
    fun create(config: VelocityConfig, maxTotal: Int): JedisPool =
        create(config, JedisPoolConfig().apply { setMaxTotal(maxTotal) })

    @JvmStatic
    fun create(config: VelocityConfig, poolConfig: JedisPoolConfig): JedisPool {
        val password = config.getRedisPassword()
        return if (!password.isNullOrEmpty()) {
            JedisPool(poolConfig, config.getRedisHost(), config.getRedisPort(), TIMEOUT_MS, password)
        } else {
            JedisPool(poolConfig, config.getRedisHost(), config.getRedisPort(), TIMEOUT_MS)
        }
    }
}
